Override favicon if the app icon is available

gayanSandamal opened the pull request:
## Description

https://github.com/user-attachments/assets/b4f04a8c-fd62-482d-95ec-a82e5b65f535

_This doesn't need any documentation update as it has been done in https://github.com/FlowFuse/node-red-dashboard/pull/1334

## Related Issue(s)

https://github.com/FlowFuse/node-red-dashboard/issues/1327

## Checklist

<!-- https://flowfuse.com/handbook/development/#defining-done -->

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass <!-- If not adding test coverage, please clarify why not? -->
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

joepavitt commented:
As discussed in 1:1, let's try not having the Dashboard icon there by default, and switching to it _if_ no `appIcon` is defined. This may help with the "flashing" occurring when first loading.

gayanSandamal commented:
> As discussed in 1:1, let's try not having the Dashboard icon there by default, and switching to it _if_ no `appIcon` is defined. This may help with the "flashing" occurring when first loading.

Noted, I'll mark this PR as draft and start working on

gayanSandamal commented:
> As discussed in 1:1, let's try not having the Dashboard icon there by default, and switching to it _if_ no `appIcon` is defined. This may help with the "flashing" occurring when first loading.

Fixed

- [x] Fixed favicon flashing issue by setting a transparent favicon as it was loading favicon from NR even we have removed the `<link type="icon">` from the `index.html`
- [x] NR favicon was loading instead of dashboard 2.0 favicon. Fixed by changing the path from `/favicon.ico -> /dashboard/favicon.ico`

When the `appIcon` is not available
https://github.com/user-attachments/assets/94feba2c-2c31-47f4-8bdc-570a6b80c89f

When the `appIcon` is available
https://github.com/user-attachments/assets/b5f957c5-7f75-4f41-ac60-e7351eaa977f
